File: frontend/src/types/registration.ts
export type ExperienceLevel = "BEGINNER" | "INTERMEDIATE" | "ADVANCED";

export interface RegistrationRequest {
  fullName: string;
  email: string;
  whatsappNumber: string;
  courseId: number;
  cohortId: number;
  experienceLevel: ExperienceLevel | null;
  referralSource: string | null;
  consentGiven: boolean;
  preferredTime: string | null;
}

export interface RegistrationResponse {
  registrationNumber: string;
  fullName: string;
  email: string;
  courseTitle: string;
  cohortName: string;
  privateTutorial: boolean;
  preferredTime: string | null;
  status: string;
}

File: backend/src/main/java/com/trainingplatform/entity/Registration.java
package com.trainingplatform.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "registrations")
public class Registration extends BaseEntity {

    @Column(name = "registration_number", nullable = false, unique = true)
    private String registrationNumber;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Column(nullable = false)
    private String email;

    @Column(name = "whatsapp_number", nullable = false)
    private String whatsappNumber;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_id", nullable = false)
    private Course course;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cohort_id", nullable = false)
    private Cohort cohort;

    @Enumerated(EnumType.STRING)
    @Column(name = "experience_level", length = 20)
    private ExperienceLevel experienceLevel;

    @Column(name = "referral_source")
    private String referralSource;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private RegistrationStatus status = RegistrationStatus.NEW;

    @Column(name = "consent_given", nullable = false)
    private boolean consentGiven = false;

    /**
     * Free-text preferred time. Required when the cohort is a private tutorial (there's no
     * fixed schedule to fall back on); optional for group cohorts, where it's just a note for
     * the admin (the cohort's actual schedule is unaffected).
     */
    @Column(name = "preferred_time")
    private String preferredTime;
}
